backend/controllers/userController.js

```diff
@@ -87,4 +91,226 @@ const loadUser = catchAsyncError(async (req, res, next) => {
+            padding: 2rem;
+            border-radius: .5rem;
+        }
+
+        .resetBtn {
+            display: inline-block;
+            text-align: center;
+            margin: 1rem auto;
+            padding: .5rem 1rem;
+            background-color: hsl(235, 40%, 53%);
+            border-radius: .4rem;
+            text-decoration: none;
+            font-weight: 600;
+            color: #fff;
+        }
+        .resetBtn:visited{
+          color:#fff;
+        }
+
+        .alternate {
+            font-size: .75rem;
+            display: grid;
+            gap: .5rem;
+        }
+    </style>
+</head>
+
+<body>
+    <h1 class="title">Forgot your Password?</h1>
+    <div class="container">
+        <p>Hi ${user.name},</p>
+        <p>Someone has requested to reset your NoteTake Password. Please click the link below to reset your password. <br/>This link will expire in 10 minutes.
+        </p>
+        <a href="${resetPasswordUrl}" class="resetBtn">Reset your
+            Password</a>
+        <p>If you didn't make this request, please ignore this email.</p>
+        <p>Thanks, <br />The NoteTake Team</p>
+        <hr />
+        <div class="alternate">
+            <p>If you are having trouble with the button above, copy and paste the URL below in your browser.</p>
+
+            <a
+                href='${resetPasswordUrl}'>${resetPasswordUrl}</a>
+            <a href="${resetPasswordUrlM}">For Mobile Users
+                (Developement
+                Reasons)</a>
+        </div>
+
+    </div>
+   
+</body>
+
+</html>`,
+  };
+  sendEmail(mailDetails);
+
+  res.status(200).json({
+    success: true,
+    message: "Forgot Password Email Sent",
+  });
+});
+
+//Reset Password Controller
+const resetPassword = catchAsyncError(async (req, res, next) => {
+  const { password, confirmPassword } = req.body;
+
+  //Hashing Token received in Params
+  const resetToken = crypto
+    .createHash("sha256")
+    .update(req.params.token)
+    .digest("hex");
+
+  //Finding Hashed resetToken the User Database with Expiry time left
+  const user = await User.findOne({
+    resetPasswordToken: resetToken,
+    resetPasswordExpire: { $gt: Date.now() },
+  });
+
+  if (!user) {
+    return next(
+      new ErrorHandler(
+        "Reset Password Token is invalid or has been expired",
+        400
+      )
+    );
+  }
+
+  if (password !== confirmPassword) {
+    return next(
+      new ErrorHandler("Password and Confirm Password are not same", 406)
+    );
+  }
+
+  user.password = password;
+  user.resetPasswordToken = undefined;
+  user.resetPasswordExpire = undefined;
+
+  //Saving Token to Database
+  await user.save();
+
+  const mailDetails = {
+    to: user.email,
+    subject: "NoteTake Password Changed",
+    // text: `Hi, ${user.name} \nPlease click below to reset your password for NoteTake. If you did not request a new password, please ignore this email.\n`,
+    html: `<!DOCTYPE html>
+    <html>
```

Comment:
Same as above create functions for each type of emails

Reply:
Please check I have made changes

- fix long html in mail options object
- by making different functions in emailTemplates for each type of email i.e. sendResetPasswordEmail()
- as per reviewers comment